src/ripple/app/paths/PathRequest.cpp

```diff
@@ -585,10 +604,20 @@ PathRequest::findPaths (std::shared_ptr<RippleLineCache> const& cache, int const
         }
     }
 
+    // Charge cost is based on source currencies used.
+    double const min_cost = 50;
+    double const max_cost = 400;
+    Resource::Charge c {static_cast<int>(
+        std::min(max_cost, std::max(
+            min_cost, std::pow(sourceCurrencies.size(), 2)))),
+                "path update"};
+    consumer_.charge(c);
+
```

Comment:
@miguelportilla I screwed up the check in the `static_assert`. I'm checking to see if the square would overflow, but I do it in a calculation that would overflow! We know `ceil(sqrt(std::numeric_limits<int>::max()))` is 46341. Let's just `static_assert(maxCost < 46341,"")` with a comment for where the magic number came from.